core/src/main/java/com/scalar/db/transaction/consensuscommit/AbstractMutationComposer.java

```diff
@@ -42,4 +43,12 @@ protected Optional<Key> getClusteringKey(Operation base, TransactionResult resul
       return base.getClusteringKey();
     }
   }
+
+  protected Key getPartitionKey(Operation base, TransactionResult result) {
+    if (base instanceof ScanAll) {
+      return result.getPartitionKey().get();
+    } else {
+      return base.getPartitionKey();
+    }
+  }
```

Comment:
When a `ScanAll` operation triggers a record recovery, since it does not hold the partition key, we need to use the partition key from the result.